docs/dev/developer-guides/contracts/contract-verification.md

```diff
@@ -1,2 +1,61 @@
-## Verify smart contracts
+## Verify smart contracts by using block explorer UI
 Now that you have learned how to deploy your contracts, you might be interested to know how you can **verify** the contracts deployed on zkSync using the official block explorer. All the details about how to verify contracts can be found on the [block explorer](https://v2-docs.zksync.io/api/tools/block-explorer/contract-verification.html) page.
+
+## Verify smart contracts with the hardhat-zksync-verify plugin
+
+To use the hardhat-zksync-verify plugin first add it to your project by running ```yarn add @matterlabs/hardhat-zksync-verify```
+After that, define your network configuration ("testnet" in this case):
+
+```
+networks: {
+  testnet: {
+    zksync: true,
+    url: 'https://zksync2-testnet.zksync.dev',
+    ethNetwork: 'goerli',
+  }
+}
+```
+
+By setting the `zksync` parameter to true when running verify task, you will be verifying smart contracts on the zkSync network, but since we support verify backward compatibility, you can also verify a smart contract directly on the Ethereum network within the same project. All you need to do is to specify your network with ```zksync: false``` settings and follow the rest of the [Etherscan verification instructions](https://hardhat.org/hardhat-runner/plugins/nomiclabs-hardhat-etherscan).
+
+### Parameters
+
+To verify the contract, you need to provide the contract's address by using the ```--contractAddress``` parameter: </br>
+```yarn hardhat verify --network <network> --address <contract address>```
```

Comment:
In the latest changes of the verify plugin I configured it to be used exactly like the etherscan verify command. When running a verify command, the address field is the first positional parameter that is expect next to other optional parameters. And when running the verification process programmatically, the "address" parameter is needed.

This will all be reflected in the newest documentation update.